src/Controller/Utils/StoreTrait.php

```diff
@@ -367,14 +370,65 @@ public function storeDeliveriesAction($id, Request $request, PaginatorInterface
                 $routes['import_success'], $orderManager, $deliveryManager, $orderFactory,);
         }
 
-        $sections = $this->getDoctrine()
-            ->getRepository(Delivery::class)
-            ->getSections($store);
+        $deliveryRepository = $this->getDoctrine()->getRepository(Delivery::class);
+        $qb = $deliveryRepository->createQueryBuilderWithTasks();
+
+        $filters = [
+            'enabled' => false,
+            'query' => null,
+            'range' => [null, null],
+        ];
+
+        if ($request->query->get('q')) {
+
+            $filters['enabled'] = true;
+            $filters['query'] = $request->query->get('q');
+
+            // Redirect startin with #
+            if (str_starts_with($filters['query'], '#')) {
+                return $this->redirectToRoute($this->getDeliveryRoutes()['view'], ['id' => intval(trim($filters['query'], '#'))]);
+            }
```

Comment:
Problems:
- The delivery may not exist. 
- Store A may try to access a delivery belonging to store B. Access would still be denied, but this can be annoying.